lib/scnnr/connection.rb

```diff
@@ -31,22 +36,40 @@ def send_json(data)
     def send_request
       block = block_given? ? Proc.new : nil
       request = build_request(&block)
-      run_request(request)
+      with_retries do
+        Net::HTTP.start(@uri.host, @uri.port, use_ssl: use_ssl?) do |http|
+          @logger&.info("Started #{@method.upcase} #{@uri}")
+          http.request(request)
+        end
+      end
     end
 
     private
 
-    def run_request(request)
-      Net::HTTP.start(@uri.host, @uri.port, use_ssl: use_ssl?) do |http|
-        @logger&.info("Started #{@method.upcase} #{@uri}")
-        http.request(request)
+    def with_retries
+      yield
+    rescue Errno::ECONNRESET => e
+      retry_count ||= 0
+
+      if retry_count < @retry_limit
+        retry_count += 1
+        @logger&.info("Retrying to connect: #{@uri}, attempt: #{retry_count}")
+
+        sleep @sleep_on_retry
+        retry
       end
+
+      raise RequestFailed, "Failed to connect: #{@uri.host} (from: #{error_location}, original: #{e.inspect})"
     end
 
     def use_ssl?
       @uri.scheme == 'https'
     end
 
+    def error_location
```

Comment:
This method is used somewhere else in order to get the name of the method + class that called it, in case of an error. But the actual return value of this method is not related to errors. Could we improve this method's name?